src/libsync/lockfilejobs.cpp

```diff
@@ -0,0 +1,199 @@
+            }
+        } else {
+            Q_EMIT finishedWithError(httpErrorCode, reply()->errorString(), {});
+        }
+    } else {
+        qCInfo(lcLockFileJob()) << "success" << path();
+        handleReply();
+        Q_EMIT finishedWithoutError();
+    }
+    return true;
+}
+
+void LockFileJob::setFileRecordLocked(SyncJournalFileRecord &record) const
+{
+    record._locked = (_lockStatus == SyncFileItem::LockStatus::LockedItem);
+    record._lockOwnerType = static_cast<int>(_lockOwnerType);
+    record._lockOwnerDisplayName = _userDisplayName;
+    record._lockOwnerId = _userId;
+    record._lockEditorApp = _editorName;
+    record._lockTime = _lockTime;
+    record._lockTimeout = _lockTimeout;
+}
+
+SyncJournalFileRecord LockFileJob::handleReply()
+{
+    const auto xml = reply()->readAll();
+
+    QXmlStreamReader reader(xml);
+
+    while (!reader.atEnd()) {
+        const auto type = reader.readNext();
+        const auto name = reader.name().toString();
+
+        switch (type) {
+        case QXmlStreamReader::TokenType::NoToken:
+        case QXmlStreamReader::TokenType::Invalid:
+        case QXmlStreamReader::TokenType::DTD:
+        case QXmlStreamReader::TokenType::EntityReference:
+        case QXmlStreamReader::TokenType::ProcessingInstruction:
+        case QXmlStreamReader::TokenType::Comment:
+        case QXmlStreamReader::TokenType::StartDocument:
+        case QXmlStreamReader::TokenType::Characters:
+        case QXmlStreamReader::TokenType::EndDocument:
+        case QXmlStreamReader::TokenType::EndElement:
+            break;
+        case QXmlStreamReader::TokenType::StartElement:
+            decodeStartElement(name, reader);
+            break;
+        }
+    }
+
+    SyncJournalFileRecord record;
+
+    if (_lockStatus == SyncFileItem::LockStatus::LockedItem) {
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::UserLock && _userDisplayName.isEmpty()) {
+            return record;
+        }
+
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::AppLock && _editorName.isEmpty()) {
+            return record;
+        }
+
+        if (_userId.isEmpty()) {
+            return record;
+        }
+
+        if (_lockTime <= 0) {
+            return record;
+        }
+
+        if (_lockTimeout <= 0) {
+            return record;
+        }
+    }
+
+    const auto relativePath = path().mid(1);
+    if (_journal->getFileRecord(relativePath, &record) && record.isValid()) {
+        setFileRecordLocked(record);
+        _journal->setFileRecord(record);
+        _journal->commit("lock file job");
+    }
+
+    return record;
+}
+
+void LockFileJob::decodeStartElement(const QString &name,
+                                     QXmlStreamReader &reader)
+{
+    if (name == QStringLiteral("lock")) {
+        const auto valueText = reader.readElementText();
+        if (valueText.isEmpty()) {
+            _lockStatus = SyncFileItem::LockStatus::UnlockedItem;
```

Comment:
@mgallien Do you need to set `_lockStatus = SyncFileItem::LockStatus::UnlockedItem;` multiple times (during initialization of class members, at the beginning of `finished()` member, and then here)? Can you assume its value is always `_ SyncFileItem::LockStatus::UnlockedItem` from the start, and then only assign it with `_ SyncFileItem::LockStatus::LockedItem` when needed? Just to reduce occurrences of `_lockStatus = SyncFileItem::LockStatus::UnlockedItem;` ?

Reply:
yes did refactor to do that
do not repeat in the xml decode method